docs/chapter4-payload-handoff-format.rst

```diff
@@ -29,9 +29,748 @@ architecture, either 32 or 64 bits.
+
+                                                      * Cell 1: npt000ss bbbbbbbb dddddfff rrrrrrrr
+                                                      * Cell 2: hhhhhhhh hhhhhhhh hhhhhhhh hhhhhhhh
+                                                      * Cell 3: llllllll llllllll llllllll llllllll
+                                                      
+                                                      Notes:
+
+                                                      * n: relocatable region flag
+                                                      * p: prefetchable (cacheable) region flag
+                                                      * t: aliased address flag
+                                                      * ss: space code
+
+                                                        * 00: configuration space
+                                                        * 01: I/O space
+                                                        * 10: 32 bit memory space
+                                                        * 11: 64 bit memory space
+
+                                                      * bbbbbbbb: The PCI bus number
+                                                      * ddddd: The device number
+                                                      * fff: The function number
+                                                      * rrrrrrrr: Register number. Not used.
+
+                                                      
+                                                      The ranges include those MMIO or IO which are already assigned to downstream devices and available for future new devices.
+   dma-ranges        O           prop-encoded-array   Define the DMA address width capability of the root bridge. If only below 4GB region was defined, the payload may assume above 4GB DMA is not
+                                                      supported.
+   ================= =========== ==================== ======================================================================================================================================================================================================================================================================================================================================================================================================================================
+
+An example showing multiple top-level PCI nodes each with its own bus range:
+
+.. code-block:: dts
+
+   
+   pci {
+      pci-rb0@c0000000 {
+      	compatible = "..."
+      	reg = <>;
+      	// properties
+      };
+
+      pci-rb1@e0000000 {
+      	compatible = "pci"
+      	bus-range = <0x24 0x4b>;
+         #address-cells = <3>;
+         #size-cells = <2>;
+
+         //non-reloc/non-prefetch/mmio, child-addr, parent-addr, length
+      	ranges = <0x82000000 0x0 0x92000000 0x0 0x92000000 0x0 0x10BC0000
```

Comment:
I found current example may be incorrect.
when address-cells=<3> and size-cells=<2>, all below examples showing the 6 elements in each range:
https://elinux.org/Device_Tree_Usage#PCI_Address_Translation
https://www.kernel.org/doc/Documentation/devicetree/bindings/pci/nvidia%2Ctegra20-pcie.txt

Please help to confirm if below is the correct understanding of ranges format:

1. addr_hi(type)
2. addr_mid(high 32bits of address)
3. addr_low(low 32bits of address)
**_4. cpu_addr(depends on cpu capability, it will be either 32bits or 64 bits)_**
5. size_hi(high 32bits of size)
6. size_low(low 32bits of size)

examples:
            ranges = <0x42000000 0 0x80000000 0x80000000 0 0x20000000
                      0x02000000 0 0xa0000000 0xa0000000 0 0x10000000
                      0x01000000 0 0x00000000 0xb0000000 0 0x01000000>;

Reply:
In today meeting we clarified the cpu_addr(parent address) can be either 1 cell or 2 cells depending on FDT root address-cells definitions. (the special case for ranges is that pci-addr/child-addr are controlled by address-cells under pci scope while cpu_addr/parent-addr is controlled by address-cells in FDT root)